plugins/inputs/monit/monit_test.go

```diff
@@ -0,0 +1,495 @@
+// Copyright 2020, Verizon
+// Licensed under the terms of the MIT License. See LICENSE file in project root for terms.
+
+package monit
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
```

Comment:
We should add some tests for common error cases:
- wrong authentication
- errors due to the "Host and network allow list"
- invalid xml

Reply:
Could you please elaborate on 'errors due to the "Host and network allow list"' testcases

Reply:
Okay got it. With allow hosts set and a different host trying to access the monit APIs I could see "connection reset by peer" error from Monit as response. Not sure how I could replicate that behavior using the httptest Server, so I am raising that error from the http client side itself using the mock HTTP client. Please let me know if you are expecting something different, or are we aligned here?